app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardCSVStep/CSVDropzoneField.js

```diff
@@ -18,52 +20,69 @@ let dropzoneRef;
 class CSVDropzoneField extends React.Component {
   constructor() {
     super();
-    bindMethods(this, ['onFileDrop', 'confirmOverwrite']);
+    this.state = { unparsedFile: null, showConfirmOverwrite: false };
+    bindMethods(this, [
+      'onFileDrop',
+      'onCancelOverwrite',
+      'onConfirmOverwrite',
+      'handleUnparsedFile'
+    ]);
   }
 
   onFileDrop(acceptedFiles, rejectedFiles) {
-    const { input, columnNames } = this.props;
-    this.confirmOverwrite(() => {
-      if (acceptedFiles && acceptedFiles.length > 0) {
-        const reader = new FileReader();
-        reader.onload = () => {
-          csv.parse(reader.result, (err, csvRows) => {
-            const rowObjects = csvRows.map(row =>
-              columnNames.reduce(
-                (rowObject, key, index) => ({
-                  ...rowObject,
-                  [key]: row[index]
-                }),
-                {}
-              )
-            );
-            input.onChange(rowObjects);
-          });
-        };
-        reader.readAsBinaryString(acceptedFiles[0]);
+    const { input: { value } } = this.props;
+    if (acceptedFiles && acceptedFiles.length > 0) {
+      // Don't bother with the confirm dialog if there is no existing value to overwrite.
+      if (!value || value.length === 0) {
+        this.handleUnparsedFile(acceptedFiles[0]);
       } else {
-        // eslint-disable-next-line no-restricted-globals
-        alert('Invalid file extension. Only .csv files are accepted.');
+        this.setState({
+          unparsedFile: acceptedFiles[0],
+          showConfirmOverwrite: true
+        });
       }
-    });
+    } else {
+      // eslint-disable-next-line no-restricted-globals
+      alert('Invalid file extension. Only .csv files are accepted.');
```

Comment:
it's interesting that `alert` is included w/ `no-restricted-globals`. I think it's OK for now - but I guess we really should use our own alert long term...

Reply:
Yeah, replacing the `alert()` is a separate p3 issue.